charts/opentelemetry-collector/values.yaml

```diff
@@ -285,6 +285,23 @@ ingress:
 #      hosts:
 #        - collector.example.com
 
+# Additional ingresses - only created if ingress.enabled is true
+# Useful for when differently annotated ingress services are required
+extraIngresses: []
```

Comment:
I'm not sure this rare use case warrants adding another section in values.yaml. The configuration interface is already pretty complicated, and will be hard to read and maintain if we keep adding configurations for any possible use case. If this particular problem can be solved with another k8s object created manually, I wouldn't like to merge this. Looking for other opinions cc @TylerHelmuth

Reply:
@dmitryax I've mixed feelings as well on this, and I may be biased by my own env but I see 2 valid use cases for this pr:
1. Many companies use at least 2 networks (corporate and cloud/multi cloud), so just 1 collector might need at least 2 ingress types
2. sub-charting the collector chart multiple times for different types of collectors for multiple use cases.

Reply:
> 2. sub-charting the collector chart multiple times for different types of collectors for multiple use cases.

Not sure I understand how this PR helps in such a use case.

Reply:
right, depending on the setup that might work with the default ingress config

Reply:
@dmitryax @Allex1 @TylerHelmuth I understand the issue with adding more sections to `values.yaml`, but certainly from my perspective this could enhance functionality of a lot of different setups. Manually deploying ingresses isn't exactly complicated but replicating the default chart ingress multiple times (labels, names, links to services etc.) manually or through IaC which instead could be fully managed in the single chart without any breaking changes to existing installations seems like a feature some might take advantage of if they see it exists, for the cost of a new default empty list section in `values.yaml`.

I am of course completely biased as this feature would help me considerably, but I do see the OpenTelemetry Collector as somewhat niche compared to other charts as a single deployment can have a huge unpredictable combination of different receivers which can benefit from customisable endpoints, so for me at least this feature completes that package and for me at least removes any external Kubernetes resources required for a production setup of OTEL. Either way, I appreciate the feedback! Cheers.

Reply:
Ok let's add this configuration